go/pkg/tree/tree.go

```diff
@@ -146,6 +161,13 @@ func buildTree(files map[string]*fileNode) *treeNode {
 			node = child
 		}
 
+		if fn.EmptyDirectoryMarker {
+			if node.Dirs == nil {
```

Comment:
I think you would create every directory in the path in the for-loop above, so this check is redundant?

Reply:
> I think you would create every directory in the path in the for-loop above, so this check is redundant?

I was a bit paranoid what if segs is empty. It's not supposed to happen, but still...